lib/getSrcForJoin.js

```diff
@@ -1,5 +1,76 @@
 'use strict';
 
+/**
+ * Find the non PR job from given workflowGragh and dest job
+ * @method findJob
+ * @param  {Object} workflowGraph       Directed graph representation of workflow
+ * @param  {String} destJobName         The dest job name to be triggered after a join
+ * @return {Array}                      List of node object consists of job name and id
+ */
+function findJobs(workflowGraph, destJobName) {
+    const jobs = new Set();
+
+    workflowGraph.edges.forEach((edge) => {
+        if (edge.dest === destJobName && edge.join) {
+            jobs.add(workflowGraph.nodes.find(node => node.name === edge.src));
+        }
+    });
+
+    return jobs;
+}
+
+/**
+ * Find the PR job from given workflowGragh and dest job
+ * Given dest job includes `PR-{PR number}:` prefix, so it need to remove prefix at first
+ * @method findPRJobs
+ * @param  {Object} workflowGraph       Directed graph representation of workflow
+ * @param  {String} destJobName         The dest job name to be triggered after a join
+ * @return {Array}                      List of node object consists of job name and id
```

Comment:
In this function, `jobs` is `Set`, not `Array`. 
I think `Set` is used instead of `Array` to prevent duplication.

Ref: https://medium.com/front-end-weekly/es6-set-vs-array-what-and-when-efc055655e1a